org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java

```diff
@@ -46,46 +52,47 @@ public void formatComment(DOMComment commentNode, XMLFormattingConstraints paren
 
 		boolean addLineSeparator = formatterDocument.hasLineBreak(leftWhitespaceOffset, start);
 		if (addLineSeparator) {
-			// Indent in case of comment being at the start of line
 			replaceLeftSpacesWithIndentation(parentConstraints.getIndentLevel(), start,
 					addLineSeparator, edits);
 		}
 		int spaceStart = -1;
 		int spaceEnd = -1;
 
-		if (isJoinCommentLines()) {
-			for (int i = commentNode.getStartContent(); i < commentNode.getEndContent(); i++) {
-				char c = text.charAt(i);
-				if (Character.isWhitespace(c)) {
-					// Whitespaces
-					if (spaceStart == -1) {
-						spaceStart = i;
-					} else {
-						spaceEnd = i;
-					}
+		for (int i = commentNode.getStartContent(); i < commentNode.getEndContent(); i++) {
+			char c = text.charAt(i);
+			if (Character.isWhitespace(c)) {
+				// Whitespaces
+				if (spaceStart == -1) {
+					spaceStart = i;
 				} else {
-					int contentStart = i;
-					while (i < commentNode.getEnd() + 1 && !Character.isWhitespace(text.charAt(i + 1))) {
-						i++;
-					}
-					int contentEnd = i;
-					availableLineWidth -= (contentEnd + 1 - contentStart);
-					if (availableLineWidth <= 0) {
-						if (spaceStart != -1) {
-							// Add new line when the comment extends over the maximum line width
-							replaceLeftSpacesWithIndentation(parentConstraints.getIndentLevel(), contentStart,
-									true, edits);
-							int indentSpaces = (getTabSize() * parentConstraints.getIndentLevel());
-							availableLineWidth = getMaxLineWidth() - indentSpaces - (contentEnd + 1 - contentStart);
-						}
-					} else {
-						replaceSpacesWithOneSpace(spaceStart, spaceEnd, edits);
-						availableLineWidth--;
-						spaceStart = -1;
-						spaceEnd = -1;
-					}
+					spaceEnd = i;
+				}
+			} else {
+				// Ensure the edit is within the selected range
+				if (startRange != -1 && endRange != -1 && (startRange > spaceStart || endRange < spaceEnd)) {
+					return;
```

Comment:
Have we have some tests with format with range for this line code?

Reply:
I have added a test now :)